File: gofit-mobile/gofit/src/com/gofit/entities/Produit.java
package com.gofit.entities;

import com.gofit.utils.Statics;

public class Produit implements Comparable<Produit> {

    private int id;
    private String refP;
    private float prixUni;
    private int quantite;
    private String description;
    private String nomProduit;
    private String categorie;
    private String image;
    private int note;
    private int totalnote;
    private int occurence;
    private Business business;

    public Produit() {
    }

    public Produit(int id, String refP, float prixUni, int quantite, String description, String nomProduit, String categorie, String image, int note, int totalnote, int occurence, Business business) {
        this.id = id;
        this.refP = refP;
        this.prixUni = prixUni;
        this.quantite = quantite;
        this.description = description;
        this.nomProduit = nomProduit;
        this.categorie = categorie;
        this.image = image;
        this.note = note;
        this.totalnote = totalnote;
        this.occurence = occurence;
        this.business = business;
    }

    public Produit(String refP, float prixUni, int quantite, String description, String nomProduit, String categorie, String image, int note, int totalnote, int occurence, Business business) {
        this.refP = refP;
        this.prixUni = prixUni;
        this.quantite = quantite;
        this.description = description;
        this.nomProduit = nomProduit;
        this.categorie = categorie;
        this.image = image;
        this.note = note;
        this.totalnote = totalnote;
        this.occurence = occurence;
        this.business = business;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getRefP() {
        return refP;
    }

    public void setRefP(String refP) {
        this.refP = refP;
    }

    public float getPrixUni() {
        return prixUni;
    }

    public void setPrixUni(float prixUni) {
        this.prixUni = prixUni;
    }

    public int getQuantite() {
        return quantite;
    }

    public void setQuantite(int quantite) {
        this.quantite = quantite;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getNomProduit() {
        return nomProduit;
    }

    public void setNomProduit(String nomProduit) {
        this.nomProduit = nomProduit;
    }

    public String getCategorie() {
        return categorie;
    }

    public void setCategorie(String categorie) {
        this.categorie = categorie;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public int getNote() {
        return note;
    }

    public void setNote(int note) {
        this.note = note;
    }

    public int getTotalnote() {
        return totalnote;
    }

    public void setTotalnote(int totalnote) {
        this.totalnote = totalnote;
    }

    public int getOccurence() {
        return occurence;
    }

    public void setOccurence(int occurence) {
        this.occurence = occurence;
    }

    public Business getBusiness() {
        return business;
    }

    public void setBusiness(Business business) {
        this.business = business;
    }


    @Override
    public String toString() {
        return "Produit : " +
                "id=" + id
                + ", RefP=" + refP
                + ", PrixUni=" + prixUni
                + ", Quantite=" + quantite
                + ", Description=" + description
                + ", NomProduit=" + nomProduit
                + ", Categorie=" + categorie
                + ", Image=" + image
                + ", Note=" + note
                + ", Totalnote=" + totalnote
                + ", Occurence=" + occurence
                + ", Business=" + business
                ;
    }

    @Override
    public int compareTo(Produit produit) {
        switch (Statics.compareVar) {
            case "RefP":
                return this.getRefP().compareTo(produit.getRefP());
            case "PrixUni":
                return Float.compare(this.getPrixUni(), produit.getPrixUni());
            case "Quantite":
                return Integer.compare(this.getQuantite(), produit.getQuantite());
            case "Description":
                return this.getDescription().compareTo(produit.getDescription());
            case "NomProduit":
                return this.getNomProduit().compareTo(produit.getNomProduit());
            case "Categorie":
                return this.getCategorie().compareTo(produit.getCategorie());
            case "Image":
                return this.getImage().compareTo(produit.getImage());
            case "Note":
                return Integer.compare(this.getNote(), produit.getNote());
            case "Totalnote":
                return Integer.compare(this.getTotalnote(), produit.getTotalnote());
            case "Occurence":
                return Integer.compare(this.getOccurence(), produit.getOccurence());
            case "Business":
                return this.getBusiness().getNom().compareTo(produit.getBusiness().getNom());

            default:
                return 0;
        }
    }

}

File: gofit-mobile/gofit-symfony/src/Entity/Produit.php
<?php

namespace App\Entity;
use App\Repository\ProduitRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use JsonSerializable;

/**
 * Produit
 *
 * @ORM\Table(name="produit", indexes={@ORM\Index(name="FK_businessPp", columns={"FK_Businees"})})
 * @ORM\Entity(repositoryClass=ProduitRepository::class)

 */
class Produit implements JsonSerializable
{
    /**
     * @var int
     *
     * @ORM\Column(name="Produit_Id", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $produitId;

    /**
     * @Assert\NotBlank(message=" La refernce doit etre non vide")
     * @Assert\Length(
     *      min = 5,
     *      minMessage=" Entrer une reference au mini de 5 caracteres"
     *
     *     )
     * @ORM\Column(name="Ref_P", type="string", length=191, nullable=false)
     */
    private $refP;

    /**
     * @var float
     *
     * @ORM\Column(name="Prix_Uni", type="float", precision=10, scale=0, nullable=false)
     */
    private $prixUni;

    /**
     * @var int
     *
     * @ORM\Column(name="Quantite", type="integer", nullable=false)
     */
    private $quantite;
    /**
     * @Assert\NotBlank(message="description  doit etre non vide")
     * @Assert\Length(
     *      min = 7,
     *      max =5000,
     *      minMessage = "doit etre >=7 ",
     *      maxMessage = "doit etre <=5000" )
     * @ORM\Column(name="Description", type="text", length=191, nullable=false)
     */

    private $description;

    /**
     * @var string
     *
     * @ORM\Column(name="Nom_Produit", type="string", length=191, nullable=false)
     */
    private $nomProduit;

    /**
     * @var string
     *
     * @ORM\Column(name="Categorie", type="string", length=191, nullable=false)
     */
    private $categorie;

    /**
     * @var string
     *
     * @ORM\Column(name="Path_Image", type="string", length=191, nullable=true, options={"default"="NULL"})
     */
    private $pathImage;

    /**
     * @var int|null
     *
     * @ORM\Column(name="Note", type="integer", nullable=true, options={"default"="NULL"})
     */
    private $note = NULL;

    /**
     * @var int|null
     *
     * @ORM\Column(name="totalnote", type="integer", nullable=true, options={"default"="NULL"})
     */
    private $totalnote = NULL;

    /**
     * @var int|null
     *
     * @ORM\Column(name="occurence", type="integer", nullable=true, options={"default"="NULL"})
     */
    private $occurence = NULL;

    /**
     * @ORM\ManyToOne(targetEntity="App\Entity\Business", inversedBy="produits")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="FK_Businees", referencedColumnName="Business_Id")
     * })
     */
    private $business;

    public function getBusiness(): ?Business
    {
        return $this->business;
    }

    public function setBusiness(?business $business): self
    {
        $this->business = $business;

        return $this;
    }


    /**
     * @var \Business
     *
     * @ORM\ManyToOne(targetEntity="Business")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="FK_Businees", referencedColumnName="Business_Id")
     * })
     */
    private $fkBusinees;

    /**
     * @return int
     */
    public function getProduitId(): int
    {
        return $this->produitId;
    }

    /**
     * @param int $produitId
     */
    public function setProduitId(int $produitId): void
    {
        $this->produitId = $produitId;
    }

    /**
     * @return string|null
     */
    public function getRefP(): ?string
    {
        return $this->refP;
    }

    /**
     * @param string $refP
     */
    public function setRefP(string $refP): void
    {
        $this->refP = $refP;
    }


    public function getPathImage()
    {
        return $this->pathImage;
    }


    public function setPathImage($pathImage): self
    {
        $this->pathImage = $pathImage;
        return $this;
    }

    /**
     * @return float|null
     */
    public function getPrixUni(): ?float
    {
        return $this->prixUni;
    }

    /**
     * @param float $prixUni
     */
    public function setPrixUni(float $prixUni): void
    {
        $this->prixUni = $prixUni;
    }

    /**
     * @return int|null
     */
    public function getQuantite(): ?int
    {
        return $this->quantite;
    }

    /**
     * @param int $quantite
     */
    public function setQuantite(int $quantite): void
    {
        $this->quantite = $quantite;
    }

    /**
     * @return string|null
     */
    public function getDescription(): ?string
    {
        return $this->description;
    }

    /**
     * @param string $description
     */
    public function setDescription(string $description): void
    {
        $this->description = $description;
    }

    /**
     * @return string|null
     */
    public function getNomProduit(): ?string
    {
        return $this->nomProduit;
    }

    /**
     * @param string $nomProduit
     */
    public function setNomProduit(string $nomProduit): void
    {
        $this->nomProduit = $nomProduit;
    }




    /**
     * @return string int|null
     */
    public function getCategorie(): ?string
    {
        return $this->categorie;
    }

    /**
     * @param string $categorie
     */
    public function setCategorie(string $categorie): void
    {
        $this->categorie = $categorie;
    }


    /**
     * @return int|null
     */
    public function getNote(): ?int
    {
        return $this->note;
    }

    /**
     * @param int|null $note
     */
    public function setNote(?int $note): void
    {
        $this->note = $note;
    }

    /**
     * @return int|null
     */
    public function getTotalnote(): ?int
    {
        return $this->totalnote;
    }

    /**
     * @param int|null $totalnote
     */
    public function setTotalnote(?int $totalnote): void
    {
        $this->totalnote = $totalnote;
    }

    /**
     * @return int|null
     */
    public function getOccurence(): ?int
    {
        return $this->occurence;
    }

    /**
     * @param int|null $occurence
     */
    public function setOccurence(?int $occurence): void
    {
        $this->occurence = $occurence;
    }

    /**
     * @return \Business|null
     */
    public function getFkBusinees(): ?Business
    {
        return $this->fkBusinees;
    }


    public function setFkBusinees(?Business $fkBusinees): self
    {
        $this->fkBusinees = $fkBusinees;
        return $this;
    }

    public function jsonSerialize(): array
    {
        return array(
            'id' => $this->produitId,
            'refP' => $this->refP,
            'prixUni' => $this->prixUni,
            'quantite' => $this->quantite,
            'description' => $this->description,
            'nomProduit' => $this->nomProduit,
            'categorie' => $this->categorie,
            'image' => $this->pathImage,
            'note' => $this->note,
            'totalnote' => $this->totalnote,
            'occurence' => $this->occurence,
            'business' => $this->fkBusinees

        );
    }

    public function setUp($refP, $prixUni, $quantite, $description, $nomProduit, $categorie, $image, $note, $totalnote, $occurence, $business)
    {
        $this->refP = $refP;
        $this->prixUni = $prixUni;
        $this->quantite = $quantite;
        $this->description = $description;
        $this->nomProduit = $nomProduit;
        $this->categorie = $categorie;
        $this->pathImage = $image;
        $this->note = $note;
        $this->totalnote = $totalnote;
        $this->occurence = $occurence;
        $this->fkBusinees = $business;

    }
}
